From 0f4aa527ebddbe32632db5517dd4b98c1371c072 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Timo=20Ter=C3=A4s?= <timo.teras@iki.fi>
Date: Wed, 19 Oct 2016 15:35:45 +0300
Subject: [PATCH 3/5] bgpd: honor disable-connected-check option with next hop
 tracking

Make bgpd ignore connected state again if configured to do so.
---
 bgpd/bgp_fsm.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/bgpd/bgp_fsm.c b/bgpd/bgp_fsm.c
index 2e07d93..7da63ea 100644
--- a/bgpd/bgp_fsm.c
+++ b/bgpd/bgp_fsm.c
@@ -716,7 +716,8 @@ bgp_start (struct peer *peer)
     }
 
   /* Register to be notified on peer up */
-  if ((peer->ttl == 1) || (peer->gtsm_hops == 1))
+  if ((peer->ttl == 1 || peer->gtsm_hops == 1) &&
+      ! CHECK_FLAG (peer->flags, PEER_FLAG_DISABLE_CONNECTED_CHECK))
     connected = 1;
 
   bgp_find_or_add_nexthop(family2afi(peer->su.sa.sa_family), NULL, peer,
-- 
2.10.1

